src/test/java/htsjdk/samtools/util/IOUtilTest.java

```diff
@@ -582,6 +582,25 @@ public void testCompressionLevel(final Path file, final String extension, final
         }
     }
 
+    @DataProvider
+    public Object[][] getExtensions(){
+        return new Object[][]{{".gz"}, {".bfq"}, {".txt"}};
+    }
+
+    @Test(dataProvider = "getExtensions")
+    public void testReadWriteJimfs(String extension) throws IOException {
+        final Path jmfsRoot = inMemoryFileSystem.getRootDirectories().iterator().next();
+        final Path tmp = Files.createTempFile(jmfsRoot, "test", extension);
+        final String expected = "lorem ipswitch, nantucket, bucket";
+        try( final Writer out = IOUtil.openFileForBufferedWriting(tmp)){
+            out.write(expected);
+        }
+        try( final BufferedReader in = IOUtil.openFileForBufferedReading(tmp)){
```

Comment:
can you somehow check that the zipped files are **really** zipped and that the plaintext ones are really plaintext?

Reply:
done